### ISSUE TITLE
App-API trait object without DYN

### ISSUE THREAD

**Issue by reesmanp** [kubos/kubos#460]
**Description**
During compile of my app which pulls in `kubos-app = { path = "../kubos/apis/app-api/rust" }` the compilation fails with:
`trait objects without an explicit 'dyn' are deprecated`
```
kubos/apis/app-api/rust/src/framework.rs:103:15
    |
103 |     handler: &AppHandler,
    |               ^^^^^^^^^^ help: use `dyn`: `dyn AppHandler`
    |
note: lint level defined here
/kubos/apis/app-api/rust/src/framework.rs:20:9
```

**System Details**
- Compiling on MacOS Mojave
- Latest KubOS master pull

**Severity**
- High - No workaround, impacting development

**To Reproduce**
1. Add KubOS-app as dependency
2. Build with `cargo build --release`

**Comment by plauche** [comment]
@reesmanp which version of `rust` are you compiling with? We have our repo pinned at `1.36.0`, but if you are including one of our libraries from outside of the repo, then you may be using a different version.

**Comment by reesmanp** [comment]
I have `1.37.0` installed. Checked via `rustc --version`.

**Comment by plauche** [comment]
@reesmanp it looks like this [warning](https://github.com/rust-lang/rust/pull/61203/) is new as of [1.37.0](https://github.com/rust-lang/rust/blob/master/RELEASES.md#compatibility-notes). 

Unfortunately the `rust-toolchain` file does not appear to pin the version for you when using our libraries as an external dependency. The quickest fix for you right now will be to use rust `1.36.0`.

**Comment by reesmanp** [comment]
Agreed that the short term fix is to jump back to the previous version, however this is still a bug within KubOs as I am assuming that the pinned version will not stay at `1.36.0` forever.

**Comment by plauche** [comment]
Yup you are correct, it will need to be fixed in the code base. I'll leave this issue open until then. I'd also like to investigate the `rust-toolchain` not applying when used in a dependency. It would be nice to not have to constantly keep up with warnings and errors introduced with each new rust version.
